system-helper/flatpak-system-helper.c

```diff
@@ -630,6 +632,42 @@ handle_deploy (FlatpakSystemHelper   *object,
   return TRUE;
 }
 
+static gboolean
+handle_cancel_pull (FlatpakSystemHelper   *object,
+                    GDBusMethodInvocation *invocation,
+                    guint                  arg_flags,
+                    const gchar           *arg_installation,
+                    const gchar           *arg_src_dir)
+{
+  OngoingPull *ongoing_pull;
+  g_autoptr(FlatpakDir) system = NULL;
+  g_autoptr(GError) error = NULL;
+
+  g_debug ("CancelPull %s %u %s", arg_installation, arg_flags, arg_src_dir);
+
+  system = dir_get_system (arg_installation, get_sender_pid (invocation), &error);
+  if (system == NULL)
+    {
+      g_dbus_method_invocation_return_gerror (invocation, error);
+      return TRUE;
+    }
+
+  ongoing_pull = take_ongoing_pull_by_dir (arg_src_dir);
+  if (ongoing_pull == NULL)
+    {
+      g_set_error (&error, G_DBUS_ERROR, G_DBUS_ERROR_FAILED,
+                   "Cannot find ongoing pull to cancel at %s", arg_src_dir);
+      g_dbus_method_invocation_return_gerror (invocation, error);
+      return TRUE;
+    }
+
```

Comment:
We need to verify that the peer uid and ongoing_pull->uid matches here too.

Reply:
Addressed.